librz/bin/format/coff/coff.c

```diff
@@ -47,32 +47,29 @@ RZ_API ut64 rz_coff_perms_from_section_flags(ut32 flags) {
 	return r;
 }
 
-RZ_API char *rz_coff_symbol_name(struct rz_bin_coff_obj *obj, void *ptr) {
-	char n[256] = { 0 };
-	int len = 0, offset = 0;
-	union {
-		char name[8];
-		struct {
-			ut32 zero;
-			ut32 offset;
-		};
-	} *p = ptr;
+/*
+ * Resolve a coff name to a C string.
+ * \param ptr buffer of at least 8 bytes
+ */
+RZ_API char *rz_coff_symbol_name(struct rz_bin_coff_obj *obj, const ut8 *ptr) {
+	rz_return_val_if_fail(obj && ptr, NULL);
+	ut32 zero = rz_read_at_le32(ptr, 0);
+	ut32 offset = rz_read_at_le32(ptr, 4);
```

Comment:
i saw in the coff config that coff supports a BE format. are we sure 100% that this shouldn't be ble ?

Reply:
Good point, changed to respect the binary endian. Seems to be really exotic though, I wonder what a good source for a test binary would be.